src/pypg/transcode.py

```diff
@@ -55,6 +53,18 @@ def _resolve_handler(cls, obj_type, overrides: dict[type, type[Self]]):
         return cls[obj_type]
 
 
+class MonotonicID:
+    def __init__(self):
+        self.objects: dict[object, int] = {}
+
+    def __call__(self, obj: object) -> int:
+        try:
+            return self.objects[id(obj)]
+        except KeyError:
+            self.objects[id(obj)] = (objid := len(self.objects))
+            return objid
```

Comment:
nit: seems like a case to use `setdefault`

Reply:
unless I'm missing something that will not produce the correct result. set_default takes a constant, and we need to re-evaluate len(self.objects) each time.